File: tests/adapters/go/goldens/comments/strip_all.go
// … comment omitted
package main

import (
	"fmt"
	"regexp"
	"time"
)

// … docstring omitted

const ModuleVersion = "1.0.0" // … docstring omitted


// … docstring omitted

type User struct {
	ID      int    // … comment omitted
	Name    string // … comment omitted
	Email   string // … comment omitted
	Profile *Profile // … comment omitted
}

// … docstring omitted

type Profile struct {
	Bio    string
	Avatar string
}

// … docstring omitted

type CommentedService struct {
	config interface{} // … comment omitted
	logger interface{} // … comment omitted
}

// … docstring omitted

func NewCommentedService(config, logger interface{}) *CommentedService {
	service := &CommentedService{
		config: config,
		logger: logger,
	}

	// … comment omitted

	return service
}

// … docstring omitted

func ProcessUser(userData *User) (*User, error) {
	// … comment omitted
	if userData == nil {
		return nil, fmt.Errorf("user data is required")
	}

	// … comment omitted
	validationResult := validateUser(userData)
	if !validationResult.IsValid {
		// … comment omitted
		fmt.Printf("Validation failed: %v\n", validationResult.Errors)
		return nil, fmt.Errorf("validation failed")
	}

	// … comment omitted
	transformed := transformUserData(userData)

	// … comment omitted
	saved, err := saveUser(transformed)
	if err != nil {
		return nil, err
	}

	return saved, nil // … comment omitted
}

func validateUser(userData *User) ValidationResult {
	// … comment omitted
	errors := make([]string, 0)

	// … comment omitted
	if userData.Name == "" {
		errors = append(errors, "Name is required") // … comment omitted
	}

	if userData.Email == "" {
		errors = append(errors, "Email is required")
	}

	// … comment omitted
	emailRegex := regexp.MustCompile(`^[^\s@]+@[^\s@]+\.[^\s@]+$`)
	if userData.Email != "" && !emailRegex.MatchString(userData.Email) {
		errors = append(errors, "Invalid email format")
	}

	return ValidationResult{
		IsValid: len(errors) == 0,
		Errors:  errors,
	}
}

// … comment omitted
func transformUserData(userData *User) *User {
	// … comment omitted
	return &User{
		ID:    generateUserID(),           // … comment omitted
		Name:  trimString(userData.Name),  // … comment omitted
		Email: toLowerCase(userData.Email), // … comment omitted
		Profile: userData.Profile,          // … comment omitted
	}
}

// … comment omitted
func generateUserID() int {
	// … comment omitted
	return int(time.Now().Unix())
}

// … comment omitted
func saveUser(user *User) (*User, error) {
	// … comment omitted
	if user != nil {
		fmt.Printf("Saving user: %d\n", user.ID)
	}

	// … comment omitted

	return user, nil // … comment omitted
}

// … docstring omitted

func ProcessString(input string) string {
	// … comment omitted
	if input == "" {
		return "" // … comment omitted
	}

	// … comment omitted
	trimmed := trimString(input)
	lowercase := toLowerCase(trimmed)
	cleaned := removeSpecialChars(lowercase)

	return cleaned // … comment omitted
}

// … comment omitted
func undocumentedHelper() {
	// … comment omitted
	data := "helper data"

	// … comment omitted
	fmt.Println(data) // … comment omitted
}

// … docstring omitted

type ValidationResult struct {
	IsValid bool     // … comment omitted
	Errors  []string // … comment omitted
}

// … docstring omitted

type ServiceConfig struct {
	Timeout int    // … comment omitted
	Retries int    // … comment omitted
	BaseURL string // … comment omitted
}

// … docstring omitted

var DefaultConfig = ServiceConfig{
	Timeout: 5000,                    // … comment omitted
	Retries: 3,                       // … comment omitted
	BaseURL: "http://localhost:3000", // … comment omitted
}

func trimString(s string) string     { return s }
func toLowerCase(s string) string    { return s }
func removeSpecialChars(s string) string { return s }

File: tests/adapters/rust/goldens/comments/strip_all.rs
// … comment omitted

use std::fmt;
use regex::Regex;

// … comment omitted
const MODULE_VERSION: &str = "1.0.0"; // … comment omitted

/// … docstring omitted
#[derive(Debug, Clone)]
struct User {
    id: i32,            // … comment omitted
    name: String,       // … comment omitted
    email: String,      // … comment omitted
    profile: Option<Profile>, // … comment omitted
}

/// … docstring omitted
#[derive(Debug, Clone)]
struct Profile {
    bio: String,
    avatar: String,
}

/// … docstring omitted
struct CommentedService {
    config: Box<dyn std::any::Any>, // … comment omitted
    logger: Option<Box<dyn std::any::Any>>, // … comment omitted
}

impl CommentedService {
    /// … docstring omitted
    fn new(config: Box<dyn std::any::Any>, logger: Option<Box<dyn std::any::Any>>) -> Self {
        let service = Self { config, logger };

        // … comment omitted

        service
    }

    /// … docstring omitted
    fn process_user(&self, user_data: User) -> Result<User, String> {
        // … comment omitted
        if user_data.name.is_empty() {
            return Err("User data is required".to_string());
        }

        // … comment omitted
        let validation_result = self.validate_user(&user_data);
        if !validation_result.is_valid {
            // … comment omitted
            eprintln!("Validation failed: {:?}", validation_result.errors);
            return Err("Validation failed".to_string());
        }

        // … comment omitted
        let transformed = self.transform_user_data(user_data);

        // … comment omitted
        let saved = self.save_user(transformed)?;

        Ok(saved) // … comment omitted
    }

    fn validate_user(&self, user_data: &User) -> ValidationResult {
        // … comment omitted
        let mut errors = Vec::new();

        // … comment omitted
        if user_data.name.is_empty() {
            errors.push("Name is required".to_string()); // … comment omitted
        }

        if user_data.email.is_empty() {
            errors.push("Email is required".to_string());
        }

        // … comment omitted
        let email_regex = Regex::new(r"^[^\s@]+@[^\s@]+\.[^\s@]+$").unwrap();
        if !user_data.email.is_empty() && !email_regex.is_match(&user_data.email) {
            errors.push("Invalid email format".to_string());
        }

        ValidationResult {
            is_valid: errors.is_empty(),
            errors,
        }
    }

    fn transform_user_data(&self, user_data: User) -> User {
        // … comment omitted
        User {
            id: self.generate_user_id(),           // … comment omitted
            name: user_data.name.trim().to_string(),  // … comment omitted
            email: user_data.email.to_lowercase(), // … comment omitted
            profile: user_data.profile,            // … comment omitted
        }
    }

    /// … docstring omitted
    fn generate_user_id(&self) -> i32 {
        // … comment omitted
        use std::time::{SystemTime, UNIX_EPOCH};
        let duration = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
        (duration.as_secs() % 1_000_000) as i32
    }

    // … comment omitted
    fn save_user(&self, user: User) -> Result<User, String> {
        // … comment omitted
        println!("Saving user: {}", user.id);

        // … comment omitted

        Ok(user) // … comment omitted
    }
}

/// … docstring omitted
fn process_string(input: &str) -> String {
    // … comment omitted
    if input.is_empty() {
        return String::new(); // … comment omitted
    }

    // … comment omitted
    let trimmed = input.trim();
    let lowercase = trimmed.to_lowercase();
    let cleaned = lowercase.chars().filter(|c| c.is_alphanumeric() || c.is_whitespace()).collect();

    cleaned // … comment omitted
}

// … comment omitted
fn undocumented_helper() {
    // … comment omitted
    let data = "helper data";

    // … comment omitted
    println!("{}", data); // … comment omitted
}

/// … docstring omitted
struct ValidationResult {
    is_valid: bool,     // … comment omitted
    errors: Vec<String>, // … comment omitted
}

/// … docstring omitted
struct ServiceConfig {
    timeout: i32,    // … comment omitted
    retries: i32,    // … comment omitted
    base_url: String, // … comment omitted
}

// … comment omitted
const DEFAULT_CONFIG: ServiceConfig = ServiceConfig {
    timeout: 5000,                    // … comment omitted
    retries: 3,                       // … comment omitted
    base_url: String::from("http://localhost:3000"), // … comment omitted
};
